client/src/components/RandomFactCard.tsx

```diff
@@ -0,0 +1,38 @@
+import React from 'react';
+import { Typography, CardContent, Card, CardMedia, Button, CardActions } from '@mui/material';
+
+interface RandomFactProps {
+  fact: string;
+}
+
+const RandomFactCard: React.FC<RandomFactProps> = ({ fact }) => {
+  return (
+    <Card sx={{ display: 'flex', flexDirection: 'column', width: 300, height: 400 }}>
+      <CardMedia
+        component="img"
+        height="150"
+        /* NOTE: The url is appended with a unique query string to
+                 circumvent caching. Since the url is the same 
+                 otherwise, browsers would only generate 1 image. */
+        image={`https://picsum.photos/300/200?${Date.now()}`}
+        alt="A randomly generated image"
+      />
+      <CardContent sx={{ maxHeight: 200, overflowY: 'auto' }}>
```

Comment:
instead of all these inline CSS, can we give them classes?

Reply:
Hey Bonnie, this is one of the biggest conflicts for me between mui v4 and v5. Mui v5 effectively [deprecated the `make/useStyles`](https://mui.com/styles/basics/) styling solution with v5 and they argue [here](https://mui.com/system/basics/) that the new alternative, the `sx` prop, is a better way of doing things.

I was a bit conflicted about this at first, but the more I use it the more it grows on me. That aside though, since they also updated most of their examples to use this styling solution, I was worried that using a `className` based convention could lead to a point of confusion for the candidates. Kindly, take a look and let me know what you think 🙏🏼

Reply:
Seems like they are moving away from classNames, still provide the ability for it. Likely for backward compatibility. `sx` does have access to our Theme, honestly I think we can leverage theme files more in CAP projects with overrides and use SX when we need to override the override (lol if that makes sense).

In that case, we could see less in-line styles in the components. 

@bonnieli Let me know what you think here

Reply:
Hi @moffatethan, good points there! I definitely agree about the theme overrides, however, I feel like promoting/encouraging this idea would require a little bit more work to be effective - I've experimented conveying this idea to candidates at varying depths and I found it difficult to get on the same page often. I've been thinking about a better way of showcasing this for a while...

One of the ideas was to add a "theme inspector" page to the starter code where we showcase different aspects of the theme such as [typography variants](https://mui.com/components/typography/#component) [and computed palette colors](https://mui.com/customization/palette/#default-values) as well as display variants of some of the most common components, Metronic is inspiring when it comes to this, such as [buttons](https://preview.keenthemes.com/metronic/react/demo1/google-material/inputs/buttons), [text fields](https://preview.keenthemes.com/metronic/react/demo1/google-material/inputs/text-fields), etc. to really show the component "arsenal" of material ui out of the box with just theme overrides.

I believe if we provide the candidates a tool where they can play with the theme, hands-on, and see the immediate impact of their changes, they would be more inclined to get familiar with the theme and more confident in using it more often.

I don't know, is this something we would be interested in? @rajivtitus @bonnieli

Reply:
@ugur-akin Yes, having a playground where candidates can interact with the theme first hand will increase their familiarity and they will feel a lot more comfortable leveraging the theme's functionality. Maybe we can implement something as such in the near future? @bonnieli

Reply:
I don't think that's necessary, their is already existing theme customizers in MUI see https://bareynol.github.io/mui-theme-creator/ 

I'll try this out in my next cohort with using theme overrides and see if it helps keep things consistent with the overall team. My thought is if someone can get a grasp on customizing the theme via the theme file then it might help for the others to just focus on the core components without having to worry about the basic theming aspects of it.

Reply:
> But I don't think we need to spend time creating one since it already exists.

Yes! I looked for a similar resource for so long and since I couldn't find anything I thought we would have to put something together. It only supports v`^4.11` currently though, that's something to keep in mind because there have been some [minor changes to the theme structure in v5](https://mui.com/guides/migration-v4/#theme-structure).

Reply:
Good point on the version, not a major difference between the two as far as I can see. Definitely something I can tell candidates, I don't think we need to spend time building this ourselves. I want to keep the starter code lean and let candidates use documentation. 

On the job, they won't be given this stuff out of the box. They'll need to search for it, so we should keep things lean. 👍🏻 

Thinking about this more, I want them to learn how to navigate documentation effectively and not rely on being provided all the details of the theme. 

`sx` is fine right now, seems like it's the preferred approach in v5 now. 

Let's focus on the other comments so we can get this approved and merged in 💪🏻

Reply:
hmmm interesting! that's good to know. this is something we can all think about. i wasn't aware they're trying to get rid of `useStyles`. I'll read more into it before developing my opinion here. 

For now its fine to go with what you have so far :)

Reply:
I created a new issue from this comment where we can further discuss the direction we want to go with this as well as if we want to take any actions 👍🏼